packages/aztec.js/src/encoder/outputCoder.js

```diff
@@ -197,33 +224,19 @@ outputCoder.encodeNotes = (notes, isOutput) => {
  * together
  */
 outputCoder.encodeOutputNote = (note) => {
-    const encoded = Array(7).fill();
-
-    // boolean to mark whether custom metadata exists. Used to automatically update note encodings if present
-    let isMetadataPresent;
-    let noteDataLength;
-    let noteLength;
-    if (note.metadata) {
-        encoded[7] = note.metadata.slice(2);
-        isMetadataPresent = 1;
-
-        const metaDataSize = 0x137;
-        noteDataLength = (0x20 * 2 + isMetadataPresent * metaDataSize).toString(16);
-        noteLength = (0x20 * 4 + 0x20 * 2 + isMetadataPresent * metaDataSize).toString(16);
-    } else {
-        encoded[7] = secp256k1.compress(note.ephemeral.getPublic()).slice(2);
-        noteLength = 'e1';
-        noteDataLength = '61';
-    }
+    const metaDataSize = parseInt(note.metaData.slice(2).length / 2, 10);
```

Comment:
The encoding functions suffer from the same thing as the decoding functions: the only difference seems to be that some computed values need to take into account the length of the metadata. This is something we can derive internally to the function (with a simple if(note.metadata)).

Reply:
The encoding functions are working from a slightly different input to the decoding functions, which means I don't believe the suggestions that work for the decoding functions will work for the encoding functions. 

In the decoding case, the function takes already correctly encoded notes i.e. `inputNotes` don't have any `metaData`, whilst `outputNotes` do have the `metaData`. So discriminating between input and output notes within the same decode function, based on the presence of `metaData` (and it's resulting impact on the size of the note) works. 

However in the encoding case, the function takes in unencoded notes still in the raw note schema format. At this stage, `inputNotes` and `outputNotes` are undifferentiated and, for example, both have `metaData`. So, discriminating between them based on the presence of `metaData` won't work, and there has to be a custom encode function for each type. 

This encoding behaviour could be solved by defining a different note schema for `inputNotes` and `outputNotes`, but I think it's simpler to stick with one schema for all notes.